backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointServiceTest.java

```diff
@@ -463,6 +463,136 @@ void testEndpointUpdates() {
         assertEquals("not-so-secret-anymore", attrSingleUpdated.getString("secret_token"));
     }
 
+    @Test
+    void testEndpointTypeQuery() {
+        String tenant = "limiter";
+        String userName = "user";
+        String identityHeaderValue = TestHelpers.encodeIdentityInfo(tenant, userName);
+        Header identityHeader = TestHelpers.createIdentityHeader(identityHeaderValue);
+
+        mockServerConfig.addMockRbacAccess(identityHeaderValue, MockServerClientConfig.RbacAccess.FULL_ACCESS);
+
+        // Add webhook
+        WebhookProperties properties = new WebhookProperties();
+        properties.setMethod(HttpType.POST);
+        properties.setDisableSslVerification(false);
+        properties.setSecretToken("my-super-secret-token");
+        properties.setUrl(String.format("https://%s", mockServerConfig.getRunningAddress()));
+
+        Endpoint ep = new Endpoint();
+        ep.setType(EndpointType.WEBHOOK);
+        ep.setName("endpoint to find");
+        ep.setDescription("needle in the haystack");
+        ep.setEnabled(true);
+        ep.setProperties(properties);
+
+        Response response = given()
+                .header(identityHeader)
+                .when()
+                .contentType(JSON)
+                .body(Json.encode(ep))
+                .post("/endpoints")
+                .then()
+                .statusCode(200)
+                .contentType(JSON)
+                .extract().response();
+
+        JsonObject responsePoint = new JsonObject(response.getBody().asString());
+        responsePoint.mapTo(Endpoint.class);
+        assertNotNull(responsePoint.getString("id"));
+
+        // Add Camel
+        CamelProperties camelProperties = new CamelProperties();
+        camelProperties.setDisableSslVerification(false);
+        camelProperties.setSecretToken("my-super-secret-token");
+        camelProperties.setUrl(String.format("https://%s", mockServerConfig.getRunningAddress()));
+        camelProperties.setExtras(new HashMap<>());
+        camelProperties.setSubType("demo");
+
+        Endpoint camelEp = new Endpoint();
+        camelEp.setType(EndpointType.CAMEL);
+        camelEp.setName("endpoint to find");
+        camelEp.setDescription("needle in the haystack");
+        camelEp.setEnabled(true);
+        camelEp.setProperties(camelProperties);
+
+        response = given()
```

Comment:
I would prefer two tests so its clear what the responsibility of the test is.
We can probably split it in a camel and webhook test don't we?

Reply:
The purpose of the test is to actually verify that the `type` param is used correctly and that it supports both 1 type or multiple types.

I could probably make the test more focused If you consider it is necesary

Reply:
Parametrized the test, let me know what you think.

Reply:
That's much better :) thanks